flake.nix

```diff
@@ -13,57 +12,63 @@
   outputs = {
     self,
     nixpkgs,
-    flake-utils,
     rust-overlay,
     ...
   }: let
-    gitRev = self.rev or self.dirtyRev or null;
-  in
-    flake-utils.lib.eachDefaultSystem (system: let
-      pkgs = import nixpkgs {
+    inherit (nixpkgs) lib;
+    systems = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
+    eachSystem = lib.genAttrs systems;
+    pkgsFor = eachSystem (system:
+      import nixpkgs {
         inherit system;
```

Comment:
Please use `localSystem.system = system` or `localSystem = system` (I prefer the former). If you dig around in `top-level/impure.nix`, you'll see that the `systems` argument is only accepted for backwards compatibility and is soft-deprecated.